File: python/src/exercises/darts/darts.py
from math import sqrt


def score(x, y):
    dist = get_distance_from_origin(x, y)
    if dist > 10:
        return 0
    elif dist <= 10 and dist > 5:
        return 1
    elif dist <= 5 and dist > 1:
        return 5
    elif dist <= 1 and dist >= 0:
        return 10
    else:
        return -1


def get_distance_from_origin(x, y):
    sq_sum = y * y + x * x
    return sqrt(sq_sum)

File: java/src/main/java/com/ybsid/exercism/practice/Darts.java
package com.ybsid.exercism.practice;

/**
 * <a href="https://exercism.org/tracks/java/exercises/darts">...</a>
 */
class Darts {
    int score(double xOfDart, double yOfDart) {
        double z = Math.sqrt(Math.pow(xOfDart,2) + Math.pow(yOfDart,2));
        if(z>10.0){
            return 0;
        }else if (z>5.0 && z<=10.0){
            return 1;
        }else if (z>1.0 && z<=5.0){
            return 5;
        }else{
            return 10;
        }
    }
}